titus-server-federation/src/main/java/com/netflix/titus/federation/service/AggregatingJobServiceGateway.java

```diff
@@ -135,15 +138,24 @@ public AggregatingJobServiceGateway(GrpcConfiguration grpcConfiguration,
         }
         JobManagementServiceStub client = optionalClient.get();
 
-        JobDescriptor withStackName = addStackName(jobDescriptor);
+        JobDescriptor enrichedJobDescriptor;
```

Comment:
the enrichment below is not fully backwards compatible, e.g.: if the `federationJobIdCreation` flag is disabled, jobs that are cloned and already have a `JOB_ATTRIBUTES_FEDERATED_JOB_ID` set will cause jobs with duplicated ids to be created. It also would allow users to directly provide an id for reasons other than cloning, which can be dangerous

Reply:
I would recommend stripping out (sanitizing) the attribute as the first step, then conditionally add it after if the configuration (feature) flag is enabled

Reply:
This is a good point. I think this is in alignment with what @corindwyer was suggesting above. 
If we strip out the attribute from the JobDescriptor once it lands on the job coordinator, the case you are highlighting can also be addressed. However, I do see value in keeping it on the JobDescriptor to validate it was in fact minted by federation. 
I am leaning towards adding code to remove the attribute if found (irrespective of the feature flag value) and then conditionally adding it to the JobDescriptor. Does that sound good, @corindwyer and @fabiokung?